server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -738,7 +773,21 @@ public DiscoveryNodes build() {
                 maxNodeVersion = maxNodeVersion == null ? version : Version.max(maxNodeVersion, version);
             }
 
+            final long newNodeLeftGeneration;
+            if (minNodeVersion == null || minNodeVersion.before(Version.V_8_9_0)) {
+                assert this.nodeLeftGeneration == null || this.nodeLeftGeneration == 0L;
+                newNodeLeftGeneration = 0L;
+            } else if (this.nodeLeftGeneration != null) {
```

Comment:
just a clarification, this branch is only executed during deserialization, right?

Reply:
That's right, I added a comment to that effect in 92e63d4347d